### PR TITLE
DSD-963: Updating prod GH Storybook URL to reservoir/v1

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-963](https://jira.nypl.org/browse/DSD-963)

## This PR does the following:

- Updates the npm command to build the static storybook site to `build-storybook:v1`.
- Updates the directory of the build to `reservoir/v1` so that the production Github Pages URL has `/reservoir/v1`.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
Don't worry about Vercel failing. The settings need to be updated but we'll do that once v1 is actually out.

File: src/components/ButtonGroup/ButtonGroup.stories.mdx [hunk 1]
```diff
@@ -83,7 +83,7 @@ width.
 
 This component is a wrapper around the `Button` component. For accessibility
 information on the `Button` component, see the
-[Button Accessibility](https://nypl.github.io/nypl-design-system/reservoir/v0/?path=/docs/components-form-elements-button--button-with-controls#accessibility)
+[Button Accessibility](https://nypl.github.io/nypl-design-system/reservoir/v1/?path=/docs/components-form-elements-button--button-with-controls#accessibility)
```
Comment (bigfishdesign13) [inline src/components/ButtonGroup/ButtonGroup.stories.mdx:86]: Is this a change that was missed in the PR for `DSD-958`?

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr997-sujv0rr0esvcir9ondjfag1elzagknor.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/627929e5fcbc3532a7696204

File: src/components/ButtonGroup/ButtonGroup.stories.mdx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/ButtonGroup/ButtonGroup.stories.mdx:86]: Or added since that PR.

**Pull request merged**